packages/@react-stately/selection/src/SelectionManager.ts

```diff
@@ -398,7 +398,7 @@ export class SelectionManager implements MultipleSelectionManager {
           }
 
           // Add child keys. If cell selection is allowed, then include item children too.
-          if (item?.hasChildNodes && (this.allowsCellSelection || item.type !== 'item')) {
+          if (item && (item?.firstChildKey === undefined ? item?.hasChildNodes : item?.firstChildKey != null) && (this.allowsCellSelection || item.type !== 'item')) {
```

Comment:
have to account for old collections which don't have `firstChildKey` on their nodes